api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-blm/blm-final-results.repository.ts

```diff
@@ -1,35 +1,49 @@
 import { Injectable } from '@nestjs/common';
-import { Workspace } from '../ports/workspace';
-import { SandboxRunnerOutputHandler } from '../sandbox-runner-output-handler';
+import { InjectEntityManager } from '@nestjs/typeorm';
+import { EntityManager } from 'typeorm';
+import { BlmFinalResultEntity } from '@marxan/blm-calibration/blm-final-results.geo.entity';
+import { BlmPartialResultEntity } from './blm-partial-results.geo.entity';
 
 @Injectable()
-export class BlmFinalResultsRepository
-  implements SandboxRunnerOutputHandler<void> {
+export class BlmFinalResultsRepository {
+  constructor(
+    @InjectEntityManager()
+    private readonly entityManager: EntityManager,
+  ) {}
+
   cancel(): Promise<void> {
     return Promise.resolve(undefined);
   }
 
   /**
    * clear previous results, move the new ones to target
    * table - everything within transaction
-   *
-   * would be used in main BlmRunService after all runs finished
    */
-  dump(
-    workspace: Workspace,
-    scenarioId: string,
-    stdOutput: string[],
-    stdErr: string[] | undefined,
-  ): Promise<void> {
-    return Promise.resolve(undefined);
-  }
+  async saveFinalResults(scenarioId: string): Promise<void> {
+    await this.entityManager.transaction(async (txManager) => {
+      await txManager.delete(BlmFinalResultEntity, {
+        scenarioId,
+      });
+
+      const partialResults = await txManager.find(BlmPartialResultEntity, {
+        where: {
+          scenarioId,
+        },
+      });
+
+      await txManager.save(
+        BlmFinalResultEntity,
+        partialResults.map((partial) => ({
+          scenarioId: partial.scenarioId,
+          blmValue: partial.blmValue,
+          score: partial.score,
+          boundaryLength: partial.boundaryLength,
+        })),
+      );
```

Comment:
Shouldn't be a big deal as we won't have much rows there but consider using raw SQL:

```
insert into final_results fr (scenario_id, blm_value, score, boundary_length)
select scenario_id, blm_value, score, boundary_length) from partial_results pr 
where pr.scenario_id = $1
```

Reply:
Ok, I'll change it :+1: